scipy/interpolate/_fitpack_repro.py

```diff
@@ -656,47 +804,77 @@ def _make_splrep_impl(x, y, *, w=None, xb=None, xe=None, k=3, s=0, t=None, nest=
     if nest is None:
         # the max number of knots. This is set in _fitpack_impl.py line 274
         # and fitpack.pyf line 198
-        nest = max(m + k + 1, 2*k + 3)
+        # Ref: https://github.com/scipy/scipy/blob/596b586e25e34bd842b575bac134b4d6924c6556/scipy/interpolate/_fitpack_impl.py#L260-L263
+        if periodic:
+            nest = max(m + 2*k, 2*k + 3)
+        else:
+            nest = max(m + k + 1, 2*k + 3)
     else:
         if nest < 2*(k + 1):
-            raise ValueError(f"`nest` too small: {nest = } < 2*(k+1) = {2*(k+1)}.")    
+            raise ValueError(f"`nest` too small: {nest = } < 2*(k+1) = {2*(k+1)}.")
         if t is not None:
             raise ValueError("Either supply `t` or `nest`.")
 
     if t is None:
-        gen = _generate_knots_impl(x, y, w=w, k=k, s=s, xb=xb, xe=xe, nest=nest)
+        gen = _generate_knots_impl(x, y, w=w, k=k, s=s, xb=xb, xe=xe,
+                                   nest=nest, periodic=periodic)
         t = list(gen)[-1]
     else:
         fpcheck(x, t, k)
 
     if t.shape[0] == 2 * (k + 1):
         # nothing to optimize
-        _, _, c = _lsq_solve_qr(x, y, t, k, w)
+        _, _, c, _, _ = _lsq_solve_qr(x, y, t, k, w)
         return BSpline(t, c, k)
 
     ### solve ###
 
     # c  initial value for p.
     # https://github.com/scipy/scipy/blob/maintenance/1.11.x/scipy/interpolate/fitpack/fpcurf.f#L253
-    R, Y, _ = _lsq_solve_qr(x, y, t, k, w)
+    solve_for_p = False
+    if periodic:
+        # N.B. - Check _lsq_solve_qr computation
+        # of p for periodic splines
+        solve_for_p = True
+        R, Y, _, p, _ = _lsq_solve_qr(x, y, t, k, w,
+                                   periodic=periodic,
+                                   solve_for_p=solve_for_p)
+    else:
+        R, Y, _, _, _ = _lsq_solve_qr(x, y, t, k, w, periodic=periodic)
     nc = t.shape[0] -k -1
-    p = nc / R[:, 0].sum()
+    if not periodic:
+        p = nc / R[:, 0].sum()
+    else:
+        R, A1, A2, Z = R
```

Comment:
What's going on here? I cannot follow if `p` is defined or not. Please streamline this.

The whole `solve_for_p` argument and value-based number of outputs is extremely confusing, and should be fixed.